src/Components/3DV/SceneView.stories.tsx

```diff
@@ -16,6 +35,231 @@ export default {
     }
 };
 
+const defaultGizmoElementItems: TransformedElementItem = {
+    meshIds: ['tank6_LOD0.003_primitive0', 'tank6_LOD0.003_primitive1'],
+    parentMeshId: 'tank6_LOD0.003_primitive0'
+};
+
+export const Gizmo = () => {
+    const [state, dispatch] = useReducer(
+        ADT3DSceneBuilderReducer,
+        defaultADT3DSceneBuilderState
+    );
+    const [isGizmo, setIsGizmo] = useState<boolean>(true);
+
+    const setGizmoTransformItem = useCallback(
+        (gizmoTransformItem: TransformInfo) => {
+            dispatch({
+                type: SET_GIZMO_TRANSFORM_ITEM,
+                payload: gizmoTransformItem
+            });
+            console.log(
+                'new position: ',
+                gizmoTransformItem.position,
+                '\nnew rotation: ',
+                gizmoTransformItem.rotation
+            );
+        },
+        []
+    );
+
+    const setGizmoElementItem = useCallback(
+        (gizmoElementItem: TransformedElementItem) => {
+            dispatch({
+                type: SET_ELEMENT_TO_GIZMO,
+                payload: gizmoElementItem
+            });
+        },
+        []
+    );
+
+    const onGizmoChange = useCallback((event, checked?: boolean) => {
+        if (!checked) {
+            setGizmoElementItem({ parentMeshId: null, meshIds: null });
+        } else {
+            setGizmoElementItem(defaultGizmoElementItems);
+        }
+        setIsGizmo(!!checked);
+    }, []);
+
+    return (
+        <div style={{ position: 'relative' }}>
+            <div>
+                <Checkbox
+                    label="Show Gizmo and Transform"
+                    checked={isGizmo}
+                    onChange={onGizmoChange}
+                />
+            </div>
+            <div>
+                <SceneView
+                    modelUrl="https://cardboardresources.blob.core.windows.net/cardboard-mock-files/OutdoorTanks.gltf"
+                    gizmoElementItem={
+                        state.gizmoElementItem ?? defaultGizmoElementItems
+                    }
+                    setGizmoTransformItem={setGizmoTransformItem}
+                    gizmoTransformItem={state.gizmoTransformItem}
+                />
+            </div>
+            <div style={{ position: 'absolute', bottom: 0 }}>
+                {isGizmo && (
+                    <FakePanel
+                        gizmoTransformItem={state.gizmoTransformItem}
+                        setGizmoTransformItem={setGizmoTransformItem}
+                    ></FakePanel>
+                )}
+            </div>
+        </div>
+    );
+};
+
+interface IFakePanelProps {
+    gizmoTransformItem: TransformInfo;
+    setGizmoTransformItem?: (gizmoTransformItem: TransformInfo) => void;
+}
+
+const FakePanel = (props: IFakePanelProps) => {
+    const { gizmoTransformItem, setGizmoTransformItem } = props;
```

Comment:
and moving all those internal components and interface to that separate wrapper component as in my comment above?

Reply:
done